Files/DataModels/SidebarPinnedModel.cs

```diff
@@ -269,6 +269,19 @@ public async Task AddItemToSidebarAsync(string path)
                         locationItem.IconData = await thumbnail.ToByteArrayAsync();
                         locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
                     }
+                    else
+                    {
+                        using var thumbnailFallback = await res.Result.GetThumbnailAsync(
+                        Windows.Storage.FileProperties.ThumbnailMode.SingleItem,
+                        24,
+                        Windows.Storage.FileProperties.ThumbnailOptions.ResizeThumbnail);
+
+                        if (thumbnailFallback != null)
+                        {
+                            locationItem.IconData = await thumbnailFallback.ToByteArrayAsync();
+                            locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
+                        }
+                    }
                 }
                 else
```

Comment:
@BanCrash could you check what happens if you revert the above change but change this line to `if (locationItem.IconData == null)`?
[Not mandatory, PR is ok as it is, I'm just curious]